src/wallet/wallet.cpp

```diff
@@ -468,13 +468,31 @@ bool CWallet::Verify()
 
     uiInterface.InitMessage(_("Verifying wallet(s)..."));
 
+    // Keep track of each existing wallet absolute path in order to detect
+    // when a wallet is already loaded by using boost::filesystem::equivalent.
+    std::vector<fs::path> wallet_paths;
+
     for (const std::string& walletFile : gArgs.GetArgs("-wallet")) {
         if (boost::filesystem::path(walletFile).filename() != walletFile) {
             return InitError(_("-wallet parameter must only specify a filename (not a path)"));
-        } else if (SanitizeString(walletFile, SAFE_CHARS_FILENAME) != walletFile) {
+        }
+
+        if (SanitizeString(walletFile, SAFE_CHARS_FILENAME) != walletFile) {
             return InitError(_("Invalid characters in -wallet filename"));
         }
 
+        fs::path wallet_path = fs::absolute(walletFile, GetDataDir());
+
+        if (fs::exists(wallet_path)) {
```

Comment:
7c4058d disallows having wallets as symlinks whereas addffb2 resolves symlink to the concrete absolute path.